File: Fs.Units/Simple/FuelEconomy.fs
namespace Fs.Units.FuelEconomy

module Units =
    
    /// Kilometer per liter
    [<Measure>] type kml
    
    /// Mile per gallon
    [<Measure>] type mpg
    
open Units

module KilometerPerLiter =
    
    let create (x: float) = x * 1.0<kml>
    let toMilesPerGallon (x: float<kml>) = x * 2.35215<mpg> / 1.0<kml>
    
module MilePerGallon =
    
    let create (x: float) = x * 1.0<mpg>
    let toKilometersPerLiter (x: float<mpg>) = x / 2.35215<mpg> * 1.0<kml>

File: Fs.Units.CSharp/Simple/FuelEconomy.cs
// ReSharper disable once CheckNamespace
namespace Fs.Units.CSharp.FuelEconomy;

public static class MilePerGallonExtensions
{
    public static double MilePerGallonToKilometersPerLiter(this double x) => MilePerGallon.toKilometersPerLiter(x);
}

public static class KilometerPerLiterExtensions
{
    public static double KilometerPerLiterToMilesPerGallon(this double x) => KilometerPerLiter.toMilesPerGallon(x);
}